text/1003-cli-naming.rst

```diff
@@ -19,110 +19,238 @@ specified.
 
-3. The ``edgedb show-status`` command prompts another question: what
-   if another kind of "display status" command is needed?
+Vague verbs or nouns can be misleading
+--------------------------------------
 
-   With ``<group> <command>`` scheme, there would be no conflict between
-   ``edgedb server status`` and ``edgedb migration status``.
+**Concern**: the ``edgedb show-status`` command does not indicate what kind
+of status it will show. It could be showing status of the current migration,
+or of the currently running server, or of how many active connections the
+server has at the moment. Renaming it to ``edgedb show-migration-status``
+would make it longer to type and overall very verbose. With
+``<group> <command>`` scheme the solution is obvious:
+``edgedb migration status``.
 
-4. The ``<action>-<category>`` scheme does not allow category-specific
-   help sections.
+**Decision**: ``edgedb show-status`` should be renamed to
+``edgedb migration-status`` which is consistent with
+``edgedb migration-log``.
 
-   With ``<group> <command>`` scheme, top-level help sections are natural:
-   ``edgedb migration -h`` command can have a description of the recommended
-   migrations workflow, while ``edgedb server -h`` would explain to the user
-   how to work with EdgeDB servers.
+In general, vague nouns and verbs should be avoided unless
+the feature in question is dead obvious (like ``edgedb dump``,
+``edgedb query``, or ``edgedb server init``).
 
-5. ``edgedb configure`` currently implements the ``<group> <command>`` scheme;
-   it should have been ``edgedb insert-config``, ``edgedb set-config``,
-   ``edgedb reset-config``.
+This will also future-proof the CLI for new kinds of functionality that
+would use the same vague terms (e.g. ``edgedb time-machine-log``).
 
-6. The ``edgedb server`` CLI design outlined in RFC 1001 directly conflicts
-   with the ``<action>-<category>`` scheme.
+Category-specific help sections
+-------------------------------
 
-The last observation is what prompted the creation of this RFC.  We either
-need to update RFC 1001 to use the ``<action>-<category>`` scheme or
-adopt the ``<group> <command>`` for all other commands.  Combining two
-drastically different CLI designs in one tool is not an option.
+**Concern**: The ``<action>-<category>`` scheme does not allow
+category-specific help sections. With ``<group> <command>`` scheme, top-level
+help sections are natural: ``edgedb migration -h`` command can have a
+description of the recommended migrations workflow, while
+``edgedb server -h`` would explain to the user how to work with EdgeDB
+servers.
 
+**Decision**: the CLI is an unlikely place for a user to learn about new
+complex functionality like the migration subsystem. Better suited avenues
+for this are the online documentation and/or ``man`` pages, or better yet,
+``tldr``.
 
-Overview
-========
+To address discoverability of related sub-commands, top-level listing of
```

Comment:
(maybe) ACTIONABLE ITEM 3: re-order sub-command listing by functionality.

This is my personal suggestion to address discoverability of related commands that are not listed one by one because they're not adjacent in the alphabetical order. If this is hard to implement or deemed not needed, I'll edit this out.

Reply:
Big +1.

Reply:
It requires a bit of manual work, but we can do that.

As an alternative we can do something like:
```
SUBCOMMANDS:
   <alphabetical link>
   
FURTHER HELP:
   edgedb help migrations  -- more help on how to maintain migrations (not just list of commands)
   edgedb help server -- installing server and maintaining local installation (in fact this already works)
   edgedb help something-else  -- more help on how to maintain migrations
```
I'm not sure what are groups and topics are though.

Reply:
> It requires a bit of manual work, but we can do that.

Awesome.

> As an alternative we can do something like:

I think hand-sorting help can make it significantly easier to parse for humans. So I'd try that.